src/routes/UserSettings/EditProfileTab/EditProfileTab.js

```diff
@@ -159,76 +252,30 @@ export default class EditProfileTab extends Component {
       <label styleName='social-label'>Social Accounts</label>
       <SocialControl
         label='Facebook'
-        onLink={() => loginWithService('facebook')}
-        onChange={this.updateSettingDirectly('facebookUrl', false)}
-        unlinkAccount={unlinkAccount}
         provider='facebook'
-        value={facebookUrl} />
+        value={facebookUrl}
+        onLink={() => loginWithService('facebook')}
+        updateSettingDirectly={() => this.updateSettingDirectly('facebookUrl')}
```

Comment:
I chose to pass `this.updateSettingDirectly` down with the appropriate key value filled out so that in the child component we just need to do `updateSettingDirectly()(value)` instead of `updateSettingDirectly(mapSocialProviderToKey[provider])(value)`

Lmk if you'd prefer we just pass down `this.updateSettingDirectly`

Reply:
love it